api/src/main/java/org/apache/iceberg/rest/CreateNamespaceRequest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+import java.io.Serializable;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+
+/**
+ * Represents a REST request to create a namespace / database.
+ */
+public class CreateNamespaceRequest implements Serializable {
+
+  // TODO - Use protected so users can extend this for their own impls. Or an interface.
+  //        Currently anything but private causes an error.
+  private String namespaceName;
+  private Map<String, String> properties;
+
+  private CreateNamespaceRequest() {
+
+  }
+
+  private CreateNamespaceRequest(String namespaceName, Map<String, String> properties) {
+    this.namespaceName = namespaceName;
+    this.properties = properties;
+  }
+
+  /**
+   * Name of the database to create.
+   */
+  String getNamespaceName() {
+    return namespaceName;
+  }
+
+  void setNamespaceName(String name) {
+    this.namespaceName = name;
```

Comment:
Why include setters and a builder? Should this be immutable?

Reply:
I included setters and getters so that it would be a proper bean for serialization purposes.

I admittedly don't love them. Going through, I see that the Aliyun is using springboot and annotations instead directly in their code.

Open to suggestions / ideas around that. And possibly this ultimately won't wind up needing to be a bean at all (not a huge fan of beans in general though I recognize they have their time and place - though that might not be now).

Reply:
Ah, looks like this is probably needed because there is no serializer and it needs to adhere to the bean interface.

Reply:
I think this depends on how we want to specify the protocol. I prefer documenting the JSON objects that will be sent, which probably means going through and creating serializers/deserializers for each object.

Reply:
Yeah. I also prefer documenting the JSON objects. I had thought that the getters and setters were still needed for the serializers though.

But I'm relatively green with the Apache HTTP library that I chose for the reference calls, so maybe I can get rid of this.

I'll play around with it, but definitely agreed that I prefer to specify the protocol and to keep it language agnostic (e.g. so people can use another language on the server side in addition to the clarity that documenting the JSON objects brings over the Java classes).

Reply:
Admittedly, as I come from more of a Scala background (partcularly when implementing REST API clients), in some places I'm not sure what needs to be a bean and what doesn't for serialization. I've typically relied on case classes, which can pretty effortlessly be converted to and from JSON with much work, or more modern over the network serialization standards such as protobuf, thrift, or even Avro.

So some things might presently be beans that don't need to be. I'm happy to clean that up as I go along and come to be more familiar with some of these java bean related things =)